### PR TITLE
[4.0] Correct css classes of mod_menu and the Cassiopeia template's css for menu at the "menu" module position

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue # ?.

### Summary of Changes

This Pull Request (PR) removes the bootstrap class `flex-column` from the `ul` element of `mod_menu` and adds a BEM conforming class name to that element so this name can be used as selector in scss instead of `flex-column` in scss, and it corrects the scss for the header section of the Cassiopeia template so the menu looks as expected on the module position "menu", which currently is not the case.

Further removal of Bootstrap classes from the Casiopeia template is not subject of this PR but can be done in future.

It seems that the css has not been compiled after a recent change, so this PR shows more differences in templates/cassiopeia/css/template.css than those coming from this PR.

### Testing Instructions

Use current 4.0-dev branch with the default Cassiopeia frontent template and have a menu in module position `menu` which should be shown in horizontal direction, and some other menus in some side position shown in vertical direction.

I used blog testing data and moved the "Main Menu Blog" module to the `menu` position. In the module's advanced setting there is no class suffix defined for the module or the menu.

Check that the menu at the `menu` position (left beside the search box) is shown as expected with this PR applied.

Check that the look of all other things on the page has not changed after applying this PR.

Move the menu back to another position, e.g. `top-a` or `sidebar-right`, and check that it looks the same as without this PR at the same position.

### Expected result

See following picture with the menu marked with a red ellipse:
after

### Actual result

See following picture with the menu marked with a red ellipse:
before

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by richard67** [comment]
@ciar4n Can you check if this PR is ok?

**Comment by richard67** [comment]
Hmm, now it seems the menu module looks bad on other module positions.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20702">issues.joomla.org/tracker/joomla-cms/20702</a>.</sub>

**Comment by richard67** [comment]
Now it should be ok.
@ciar4n Can you check?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20702">issues.joomla.org/tracker/joomla-cms/20702</a>.</sub>

File: modules/mod_menu/tmpl/default.php [hunk 1]
```diff
@@ -86,7 +86,7 @@
 	// The next item is deeper.
 	if ($item->deeper)
 	{
-		echo '<ul class="list-unstyled small">';
+		echo '<ul class="mod-menu__list list-unstyled small">';
```
Comment (ciar4n) [inline modules/mod_menu/tmpl/default.php]: For now I would remove `mod-menu__list`. These items can be selected with simply `mod-menu ul`

File: templates/cassiopeia/scss/blocks/_header.scss [hunk 2]
```diff
@@ -53,7 +53,7 @@
     }
   }
 
-  .navbar-nav.flex-column {
+  .mod-menu.nav {
```
Comment (ciar4n) [inline templates/cassiopeia/scss/blocks/_header.scss]: Just `.mod-menu` should suffice. Currently adding `.nav` makes this useless if a class suffix is added.

File: templates/cassiopeia/scss/blocks/_global.scss [hunk 3]
```diff
@@ -33,3 +33,7 @@ small,
 dd {
   margin-bottom: 0;
 }
+
+.mod-menu.nav {
```
Comment (ciar4n) [inline templates/cassiopeia/scss/blocks/_global.scss]: Just `.mod-menu` should suffice. Currently adding `.nav` makes this useless if a class suffix is added.

File: modules/mod_menu/tmpl/default.php [hunk 4]
```diff
@@ -18,12 +18,12 @@
...
 
-// The menu class is deprecated. Use nav instead
+// The menu class is deprecated. Use mod-menu instead
 ?>
-<ul<?php echo $id; ?> class="nav flex-column <?php echo $class_sfx; ?>">
+<ul<?php echo $id; ?> class="mod-menu nav<?php echo $class_sfx; ?>">
 <?php foreach ($list as $i => &$item)
 {
-	$class = 'nav-item';
+	$class = 'mod-menu__item nav-item';
```
Comment (ciar4n) [inline modules/mod_menu/tmpl/default.php]: For now I would remove `mod-menu__item`. These items can be selected with simply `mod-menu li`

File: templates/cassiopeia/scss/blocks/_header.scss (hunk 2 above)
Comment (richard67) [reply templates/cassiopeia/scss/blocks/_header.scss]: Ok, but then it needs at line 70 below a change of `.nav li` to `.mod-menu li`, right?

**Comment by richard67** [comment]
@ciar4n Now it should be ok.

**Comment by wilsonge** [comment]
Nice work - thanks!

**Pull request merged**